src/proxy-animation.js

```diff
@@ -912,6 +906,147 @@ function autoAlignStartTime(details) {
   }
 }
 
+function unsupportedTimeline(timeline) {
+  throw new Error('Unsupported timeline class');
+}
+
+function getNormalStartRange(timeline) {
+  if (timeline instanceof ViewTimeline) {
+    return { rangeName: 'cover', offset: CSS.percent(0) };
+  }
+
+  if (timeline instanceof ScrollTimeline) {
+    return CSS.percent(0);
+  }
+
+  unsupportedTimeline(timeline);
+}
+
+function getNormalEndRange(timeline) {
+  if (timeline instanceof ViewTimeline) {
+    return { rangeName: 'cover', offset: CSS.percent(100) };
+  }
+  
+  if (timeline instanceof ScrollTimeline) {
+    return CSS.percent(100);
+  }
+
+  unsupportedTimeline(timeline);
+}
+
+function parseAnimationRange(timeline, value) {
+  const animationRange = {
+    start: getNormalStartRange(timeline),
+    end: getNormalEndRange(timeline),
+  };
+
+  if (!value)
+    return animationRange;
+
+  if (timeline instanceof ViewTimeline) {
+    // Format:
+    // <start-name> <start-offset> <end-name> <end-offset>
+    // <name> --> <name> 0% <name> 100%
+    // <name> <start-offset> <end-offset> --> <name> <start-offset>
+    //                                        <name> <end-offset>
+    // <start-offset> <end-offset> --> cover <start-offset> cover <end-offset>
+    // TODO: Support all formatting options once ratified in the spec.
+    const parts = value.split(' ');
```

Comment:
https://github.com/flackr/scroll-timeline/pull/209 adds a function `splitIntoComponentValues()` to split a string of component values into separate strings taking functions and parenthesis into account. 

If https://github.com/flackr/scroll-timeline/pull/209 is merged first, we could use this function here, avoiding issues like `cover calc(0% - 10px) cover calc(100% + 10px)` being split into `['cover', 'calc(0%', '-', '10px)', 'cover', 'calc(100%', '+', '10px)']`

But, we could always do this at a later point.

Reply:
Depending on which one gets in first, I’ll ask for your help to fix these here.

_(For my convenience it would be easiest if this one makes it in first 😝)_

Reply:
> (For my convenience it would be easiest if this one makes it in first 😝)

Let's do it that way 😀👍